File: trunk/ShadowFrameworkLite20/src/shadow/animation/SFCompositeAnimation.java
package shadow.animation;

import java.util.ArrayList;

public class SFCompositeAnimation extends ArrayList<SFAnimation> implements SFAnimation{

	private static final long serialVersionUID = (new SFCompositeAnimation()).getClass().hashCode();
	
	@Override
	public void animate(long time) {
		for (int i = 0; i < size(); i++) {
			get(i).animate(time);
		}
	}
	
	@Override
	public SFCompositeAnimation clone() {
		SFCompositeAnimation clone=new SFCompositeAnimation();
		for (SFAnimation sfAnimation : this) {
			SFAnimation animation=sfAnimation.clone();
			clone.add(animation);
		}
		// TODO yea, but in this way you are not going anywhere..bn
		return clone;
	}
	
	@Override
	public void destroy() {
		//nothing to do
	}
	
	
	@Override
	public void init() {
		//nothing to do
	}

	protected SFTweener getTweener() {
		return null;
	}

	protected void setTweener(SFTweener tweener) {
	}

	protected void applyTransform(double T) {
	}
}

File: trunk/ShadowFrameworkJS/src/shadow/animation/SFCompositeAnimation.js



function SFCompositeAnimation(){
	this.array=new Array();
}

SFCompositeAnimation.prototype["animate"]=function(time){
	for (var i = 0; i < this.size(); i++) {
		get(i).animate(time);
	} 
};


SFCompositeAnimation.prototype["size"]=function(){
	return this.array.length;
};

SFCompositeAnimation.prototype["get"]=function(i){
	return this.array[i];
};

SFCompositeAnimation.prototype["add"]=function(animation){
	return this.array.push(animation);
};
	
SFCompositeAnimation.prototype["clone"]=function(){
	var clone=new SFCompositeAnimation();
	for (var i=0;i<this.array.length;i++) {
		var animation=sfAnimation.clone();
		clone.add(animation);
	}
	return clone;
};
	
SFCompositeAnimation.prototype["destroy"]=function(){
	
};	

SFCompositeAnimation.prototype["init"]=function(){
	
};	
	
SFCompositeAnimation.prototype["getTweener"]=function(){
	return null;
};	

//	protected void setTweener(SFTweener tweener) {
//	}

//	protected void applyTransform(double T) {
//	}
